sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppImpl.java

```diff
@@ -807,6 +816,137 @@ Mono<ConnectionStringDictionaryInner> updateConnectionStrings(ConnectionStringDi
         }
     }
 
+    @Override
+    public void deploy(DeployType type, File file) {
+        deployAsync(type, file).block();
+    }
+
+    @Override
+    public Mono<Void> deployAsync(DeployType type, File file) {
+        return deployAsync(type, file, null);
+    }
+
+    @Override
+    public void deploy(DeployType type, File file, DeployOptions deployOptions) {
+        deployAsync(type, file, null).block();
+    }
+
+    @Override
+    public Mono<Void> deployAsync(DeployType type, File file, DeployOptions deployOptions) {
+        return this.pushDeployAsync(type, file, null)
+            .flatMap(result -> kuduClient.pollDeploymentStatus(result, manager().serviceClient().getDefaultPollInterval()));
+    }
+
+    @Override
+    public void deploy(DeployType type, InputStream file, long length) {
+        deployAsync(type, file, length).block();
+    }
+
+    @Override
+    public Mono<Void> deployAsync(DeployType type, InputStream file, long length) {
+        return deployAsync(type, file, length, null);
+    }
+
+    @Override
+    public void deploy(DeployType type, InputStream file, long length, DeployOptions deployOptions) {
+        deployAsync(type, file, length, null).block();
+    }
+
+    @Override
+    public Mono<Void> deployAsync(DeployType type, InputStream file, long length, DeployOptions deployOptions) {
+        return this.pushDeployAsync(type, file, length, null)
+            .flatMap(result -> kuduClient.pollDeploymentStatus(result, manager().serviceClient().getDefaultPollInterval()));
+    }
+
+    @Override
+    public KuduDeploymentResult pushDeploy(DeployType type, File file, DeployOptions deployOptions) {
+        return pushDeployAsync(type, file, deployOptions).block();
+    }
+
+    @Override
+    public Mono<KuduDeploymentResult> pushDeployAsync(DeployType type, File file, DeployOptions deployOptions) {
+        // If tier of the AppServicePlan is "FlexConsumption", use /api/publish; else, use /api/zipdeploy
+
+        // deployOptions is ignored
+        if (type != DeployType.ZIP) {
+            return Mono.error(new IllegalArgumentException("Deployment to Function App supports ZIP package."));
+        }
+        return getAppServicePlanIsFlexConsumptionMono().flatMap(appServiceIsFlexConsumptionPlan -> {
+            try {
+                if (appServiceIsFlexConsumptionPlan) {
+                    return kuduClient.pushDeployFlexConsumptionAsync(file);
+                } else {
+                    return kuduClient.pushZipDeployAsync(file)
+                        // it appears that "getDeploymentStatus" must first be called with deploymentId="latest"
+                        // this "latest" also works, if polling is done via kudu "api/deployments/{deploymentId}" API
+                        .then(Mono.just(new KuduDeploymentResult("latest")));
+                }
+            } catch (IOException e) {
+                return Mono.error(e);
+            }
+        });
+    }
```

Comment:
`pushDeploy` branches on tier.

For non-Flex, it calls zipDeploy
For Flex, it calls oneDeploy

There is actually another branching for ConsumptionPlan (non-Flex) that CLI uses "WEBSITE_RUN_FROM_PACKAGE"